yarn.lock

```diff
@@ -0,0 +1,5354 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```

Comment:
should we keep both `package-lock.json` and `yarn.lock`? seems redundant

Reply:
Errrrrm I’m really not sure to be honest. When I do `npx react-native init` and then do `npm install`, it updates `yarn.lock` _and_ generates `package-lock.json`. (`package-lock.json` isn't in the `.gitignore` that `npx react-native init` generates.)

I don't know what the right answer is here; it doesn't seem as simple as "`yarn.lock` is used by Yarn and `package-lock.json` is used by NPM, and since both package managers are supported you need to have both", given that NPM clearly interacts with `yarn.lock`. Nor does it seem as simple as "`yarn.lock` is all you need", given that NPM still generates `package-lock.json` in the presence of `yarn.lock`, and given that it's not gitignored by the React Native template.

Do you have any thoughts on this? I can investigate further otherwise; it's not something I know about.

Reply:
> When I do npx react-native init and then do npm install, it updates yarn.lock and generates package-lock.json. (package-lock.json isn't in the .gitignore that npx react-native init generates.)

I believe what is happening here is that `npx react-native init` is configured by default to use Yarn as the package manager, which generates `yarn.lock`. I ran `npx react-native init` locally for a clean project, and it created a `yarn.lock` file without a `package-lock.json` file.
It's still true that `yarn.lock` is used by Yarn and `package-lock.json` is used by NPM, they don't mix & match. So, when you run `npm install` afterward, NPM doesn't find a `package-lock.json` file and creates one. Lock files are never added to the .gitignore, so you end up with both lock files in git.

> given that NPM clearly interacts with yarn.lock

I'm also not entirely sure why NPM updates `yarn.lock` too when you run `npm install` (I noticed this in another project that had both lock files), but my best guess is that NPM sees `yarn.lock` and updates it to keep it in sync. This might be done to mitigate any possible dependency issues if another developer in the project uses Yarn instead of NPM.

> Nor does it seem as simple as "yarn.lock is all you need", given that NPM still generates package-lock.json in the presence of yarn.lock

You definitely need `package-lock.json` if you're using NPM, it will be using that and it just happens that is also keeps `yarn.lock` in sync.

That being said, I think we should stick with one package manager to avoid situations where a developer uses a different package manager than intended. I have yet to see a project where we use Yarn exclusively, so I suggest we stick with NPM here.
I believe just deleting the `yarn.lock` file is enough.

Reply:
Cool, thanks for thinking about it. I’ve removed it.